specs/NotifyDropTargetAction.md

```diff
@@ -0,0 +1,313 @@
+  Hopefully we'll be able to copy it mostly verbatim.
+  So the second audience is everyone that reads there to learn how
+  and why to use this API. 
+-->
+
+
+# Background
+These APIs will allow users to drop things such as images, text, and links into the WebView as part of a drag/drop operation.
+The reason that we need separate APIs for this with composition hosting is because we don't have an HWND to call RegisterDragDrop on.
+The hosting app needs to call RegisterDragDrop on the HWND that contains the WebView and implement IDropTarget so it can forward the
+calls (IDropTarget::DragEnter, DragMove, DragLeave, and Drop) to the WebView. Other UI frameworks have their own ways to register.
+For example, Xaml require setting event handler for DragEnter, DragOver, DragLeave, and Drop on the corresponding UIElement.
+
+Additionally, for win32, the hosting app also needs to call into IDropTargetHelper before forwarding those calls to us as documented here:
+https://docs.microsoft.com/en-us/windows/win32/api/shobjidl_core/nn-shobjidl_core-idroptargethelper
+
+For API reviewers, We want a unified API surface between COM and WinRT that works in both UWP and Win32.
+
+We could:
+1. Have one API focusing on Win32 types and require the end dev to convert from UWP types to Win32 (which we've done).
+2. Have one API focusing on UWP types and require the end dev to convert Win32 to UWP.
+3. Have two sets of methods one for Win32 and one for UWP types.
+Or we could do (1) or (2) and provide a conversion function.
+Because the conversion is simple and we have a large Win32 user base we chose (1) and provided a conversion function.
+
+
+# Description
+DragEnter, DragOver, DragLeave, and Drop are functions meant to provide a way for composition hosted WebViews to receive drop events as part of a drag/drop operation.
+It is the hosting application's responsibility to call RegisterDragDrop (https://docs.microsoft.com/en-us/windows/win32/api/ole2/nf-ole2-registerdragdrop)
+on the HWND that contains any composition hosted WebViews and to implement IDropTarget(https://docs.microsoft.com/en-us/windows/win32/api/oleidl/nn-oleidl-idroptarget)
+to receive the corresponding drop events from Ole32:
+
+- IDropTarget::DragEnter
+- IDropTarget::DragMove
+- IDropTarget::DragLeave
+- IDropTarget::Drop
+
+For other UI frameworks such as Xaml, the hosting application would set event handlers for DragEnter, DragOver, DragLeave, and Drop
+on the UIElement that contains the WebView2 element.
+
+# Examples
+## Win32
+```c++
+// Win32 Sample
+
+// Implementation for IDropTarget
+HRESULT DropTarget::DragEnter(IDataObject* dataObject,
+    DWORD keyState,
+    POINTL point,
+    DWORD* effect)
+{
+    POINT point = { cursorPosition.x, cursorPosition.y };
+    // Tell the helper that we entered so it can update the drag image and get
+    // the correct effect.
+    wil::com_ptr<IDropTargetHelper> dropHelper = DropHelper();
```

Comment:
Missing definition of DropHelper(). I'm hoping it is a member function that creates and caches a `CLSID_DragDropHelper` in a member variable.